src/extensions/extensionInterface.ts

```diff
@@ -209,31 +209,35 @@ export interface RecordingSelection {
 }
 
 export const useRecordingAnimation = (selection: RecordingSelection, selectionDispatch: RecordingSelectionDispatch) => {
-    const [lastUpdateTimestamp, setLastUpdateTimestamp] = useState(Number(new Date()))
-    const [animationFrame, setAnimationFrame] = useState(0)
+    const ref = useRef({
```

Comment:
Updated useRecordingAnimation so that it minimizes the number of state changes during idle, and prevents excessive CPU during idle.

Reply:
We want to minimize the number of state changes for the component that uses this hook. Previously we were changing the state (setAnimationFrame) with every iteration, even if the current timepoint was not changing. Now the state changes only when the current timepoint changes.